### ISSUE TITLE
Palemoon migration discussion

### ISSUE THREAD

**Issue by PikachuEXE** [tabkit/tabkit2#139]
I plan to migrate TK2 for Palemoon (instead of Firefox) since FF 57.0+ will no longer accept XUL extensions.
http://www.palemoon.org/

Palemoon is a Firefox fork which will not remove XUL extension support.
It supports windows officially and MacOS support is in beta/unofficial (download can be found in [Palemoon for Mac OSX section](https://forum.palemoon.org/viewforum.php?f=41)

Palemoon don't support some of the "new" Firefox extension (addon SDK)
So migrating to Palemoon might need some work depends on what extensions you are using

I am already using a modified version of TK2 (which fixes some issue with Palemoon) for a month
And it seems working fine so long

**Comment by PikachuEXE** [comment]
First beta version for Palemoon: `0.13.0.beta.1`
[tabkit2_0.13.0.beta.1.xpi.zip](https://github.com/tabkit/tabkit2/files/1424956/tabkit2_0.13.0.beta.1.xpi.zip)

I using this on Palemoon `27.5.1`

**Comment by ppetak** [comment]
tabkit is now on 56 somehow interfering with session manager, both vanilla and from mozilla extension, both on windows and on linux. As this is last version of firefox we could use tabkit with, would there be last fixed version on 56, or we should rather go to Palemoon straight away?

**Comment by PikachuEXE** [comment]
Firefox "stable" (which looks more like edge/unstable/development version to me)
is not supported since every new version often breaks something in extensions.
Which is why it's NOT supported and you should start trying Palemoon.

This could also be a reason why the old XUL extension are removed. 
They want to change the internal stuff (which many XUL extensions are using) without getting complaints.

**Comment by kstavert** [comment]
I love PaleMoon... started using it more than a year ago...

I stopped when I found OneTab and  OneTab did not work on PaleMoon
I always have too many tabs open and OneTab works fabulously well

I've found a very rough tab tree... not even in the same ballpark as
Tab Kit 2nd to work in FF... 

if PaleMoon had something like OneTab I'd be here full time in a flash... 

Any chance of that?

**Comment by gituser** [comment]
Offtopic:
I've tried new FF 57 and its blazing fast.
As an alternative you can use for now FF57 + TreeStyleTab extension instead of Tabkit2. 

The new version of Firefox is very responsive.

**Comment by kstavert** [comment]
I do like the new FF
And, I have/ am trying both of the tab tree thingys and neither even comes close to
how amazing TabKit 2nd is...  
I have large monitors and usually have a whole mess of tabs open...jumping from one project
to another .......so side tabs makes it really easy to read the tab - I don't need or want the tabs across the top - just takes up screen space unnecessarily... 
The four addons I use extensively:
TabKit 2nd
OneTab
LassPass
Session Manager (and the developer seems to have quit on this one, too)

**Comment by gituser** [comment]
I am completely agree with you about wasted space at the top, it's very annoying.
Would be nice to have Tabkit2 rewritten for the new engine if possible.

**Comment by PikachuEXE** [comment]
I use Vivaldi & Palemoon + TK2 at work & home
Haven't tried the new Firefox, for speed I just use Vivaldi (Chromium based)
And its dev DO listen to users' comments (hi Firefox)

I might use Firefox as backup browser but definitely not as one of the main browsers

**Comment by kstavert** [comment]
The other tab tree options in FF are really bad... and they more or less work
OneTab and a session manager are NOT available in Pale Moon

I can limp along with the poor sidebar tabs... NOT without OneTab or a Session Manager

**Comment by gituser** [comment]
I've tried palemoon: it seems to be based on very old firefox 29 or so and some sites are simply not working properly at all (e.g. scrolling doesnt work when you move your mouse).

Also, trying right now Vivaldi it seems chromium based and has built-in ability to get vertical tabs out there without any addons.

Firefox ESR is very laggy and slowing down UI a lot so can't be used anymore. 

I don't get why Firefox guys do not offer an option to make vertical tabs built-in after all these years it's a must have feature.

**Comment by PikachuEXE** [comment]
Vivaldi is not a complete replace for Firefox (with vertical bar).
Palemoon doesn't work on all websites, but most websites.

So for most website that I want to keep opened in tabs, I use Palemoon.
For those daily "open then close" or incompatible websites (which only support Webkit browsers or just don't support Palemoon), I use Vivaldi.

Works for me so far

new Firefox = backup browser like Chrome, or even worse

**Comment by gituser** [comment]
@PikachuEXE any chance you could port your tabkit2 to vivaldi?

**Comment by PikachuEXE** [comment]
I doubt Vivaldi got API to change the tab's itself...

**Comment by gituser** [comment]
@PikachuEXE I see. They also recently introduced the tree tabbar there, but it eats lots of space on my 24" monitor because of the additional panel. And far from perfect (i mean tab alignment and there is no colouring of the groups). Would be nice if they could hear from someone about TabKit2 concept and implement it as an option or plugin.

**Comment by Red-Eagle-LXIX** [comment]
@kstavert Session Manager version 0.8.1.7 is compatible with Pale Moon and works quite well (version I've been using for a while).
https://addons.mozilla.org/en-US/firefox/addon/session-manager/versions/

Sometimes it is necessary to check past versions of your other preferred addons to find Palemoon compatible versions.  Although a quick check shows no past compatible versions of OneTab.

**Comment by PikachuEXE** [comment]
@Red-Eagle-LXIX Fixed some strange bug when it's being used on Palemoon
Like inverted tab position
Could be due to slightly different browser layout/structure (Palemoon is forked a long time ago)
But nothing else is changed (and I can only test those I used everyday)

Open a new issue if there is any bug

**Comment by PikachuEXE** [comment]
Tab Kit 2 is now listed on Palemoon's addon website
http://addons.palemoon.org/extensions/

Will update README soon

**Comment by Red-Eagle-LXIX** [comment]
@PikachuEXE - Just updated to 0.13.0   - No issues to report.  And Thank you for correcting the wrong side placement issue.  Now it matches being on then right when set to the right (and vice-versa) instead of opposite sided.   Again glad to see you joining the Palemoon ranks.  Been using Tab Kit forever now.  I'm not even sure if I could ever go back to browsing without it and a Session Manager (which on PM, SessionManager tends to keep the tree more intact with less corruptions than I used to get on FF).

**Comment by clopez** [comment]
> Offtopic:
> I've tried new FF 57 and its blazing fast.
> As an alternative you can use for now FF57 + TreeStyleTab extension instead of Tabkit2 (the annoying thing though you cant hide tabs at the top).
> The new version of Firefox is very responsive.

Other than TreeStyleTab there is now [another vertical-tree-tab addon compatible with WebExtensions](https://addons.mozilla.org/en-US/firefox/addon/tree-tabs/) (this one has a search field) that works on the last version of FireFox. I still miss some features from TabKit2 like colored tab groups, but for sure FF with this is better than Chrome.

I'm skeptical about PaleMoon having the required resources to maintain their FF fork over time. Maintaining a browser is [a lot of work](http://robert.ocallahan.org/2017/12/maintaining-independent-browser-is.html). But in any case I think is great this extension moves there, as it seems to PaleMoon  will be the last XUL-based browser in active development.

**Comment by PikachuEXE** [comment]
Palemoon is forked a long time ago and no further changes are ported afterward from Firefox
It should be considered as a different browser now (although being able to use Firefox extensions)

I agree Firefox looks faster now and it might be what the dev team wants.
And I am happy to stay with Pale Moon and enjoy a stable browser (and don't have to fix new issues every year!)

**Comment by Red-Eagle-LXIX** [comment]
Palemoon has been around quite a long time now: http://www.palemoon.org/releasenotes-archived.shtml

That said there is also another fork of FF that supports both XUL and WebExtensions, Basilisk: http://basilisk-browser.org/

Forked by the same folks who do Palemoon with the intent to further develop XUL and Unified XUL Platform (UXP) which will be implemented in Palemoon as well.

**Comment by gituser** [comment]
>I'm skeptical about PaleMoon having the required resources to maintain their FF fork over time. Maintaining a browser is a lot of work. But in any case I think is great this extension moves there, as it seems PaleMoon will be the last XUL-based browser in active development.

completely agree with @clopez, even more I'd say that after quick testing of Palemoon there are many websites which are not loading correctly there or navigation is simply not working at all, whilst in top 4 browsers there is no issue.

 >Other than TreeStyleTab there is now another vertical-tree-tab addon compatible with WebExtensions (this one has a search field) that works on the last version of FireFox. I still miss some features from TabKit2 like colored tab groups, but for sure FF with this is better than Chrome.

Yes, I've tried the addon you linked of course there is a room for improvement that's for sure (to get to Tabkit2 level). Need to contact it's author and see if he's up to port these futures into his add-on. 

Is there any way to hide the tab bar at the top in new FF?

Also I'm using Vivaldi now thanks to @PikachuEXE - it's usable but not perfect either (because it's based on Chrome it's basically a memory eater, but still better than FF ESR+Tabkit (it's unusable now for me - very slow after 30-40 minutes of work)).

**Comment by PikachuEXE** [comment]
I don't think there is a perfect browser solution at the moment.
Either it's memory hungry (and lacking some features) or it's not fully compatible with most websites.
So I am using both Pale Moon and Vivaldi now for different purposes.

**Comment by Red-Eagle-LXIX** [comment]
Instead of making a new issue I figure I would ask here since this is a conversation on migration and discussion of such browsers being supported.
Is there a  possibility of having the Basilisk Browser's ID added to the Tab Kit version on APO?
It'd be nice to be able to stick with TK2 at the same version on both browsers.
This is not unheard of as APO have a few that mention if they work for both (example would be 'Config Comments' http://addons.palemoon.org/addon/config-comments/ ).
 Thanks.

**Comment by PikachuEXE** [comment]
@Red-Eagle-LXIX I think it's still in early dev
And I have no idea how to add ID for this browser
I have searched the forum https://forum.palemoon.org/viewforum.php?f=61 and can't find any instruction for it
Maybe it should *just work*?

**Comment by Red-Eagle-LXIX** [comment]
I cheated and hacked in the ID for Firefox to get 0.13 installed in Basilisk.
Yeah, I think early dev as well, but nice and stable for me so far.  Running in parallel with my Palemoon install.
Trying my to keep them as near as possible the same in regards to extensions and layouts (mostly because I am a creature of habit).

**Comment by PikachuEXE** [comment]
Use older one for now
Will submit it to Palemoon addon website later

Ensure to backup your current extension, who knows if Mozilla takes old extensions down after a long time :S

**Comment by PikachuEXE** [comment]
@GregordeLallo Just submitted both restartless extensions to Pale Moon
Should take a few days to process
Let me ping you when ready

**Comment by PikachuEXE** [comment]
@GregordeLallo https://addons.palemoon.org/addon/tabkit-mouse-gestures/
Tab Highlighter is still being processed (since I forgot to upload the file :P)

**Comment by GregordeLallo** [comment]
@PikachuEXE 
Thanks! It works perfectly.
Really appreciate the effort of keeping everything working. I wouldn't know how to survive without tabkit :(
